views/components/node-statistics-record-view.js

```diff
@@ -38,29 +39,41 @@ export default function NodeStatisticsRecordView({stat}) {
                 </div> :
                 <span className="inline-block"><i className="icon-warning color-warning"/> Peer nodes not connected</span>}
         </div>
-        {!!Object.keys(stat.oracleStatistics || {}).length && <OracleStatisticsView statistics={Object.values(stat.oracleStatistics)}/>}
+        {!!Object.keys(stat.contractStatistics || stat.oracleStatistics || {}).length && <ContractStatisticsView statistics={Object.values(stat.contractStatistics || stat.oracleStatistics)}/>}
     </div>
 }
 
-function OracleStatisticsView({statistics = []}) {
+function ContractStatisticsView({statistics = []}) {
     if (!statistics.length)
         return
 
-    return statistics.map(stat => <div key={stat.oracleId}>
-        <h4>Oracle <AccountAddress account={stat.oracleId}/></h4>
+    return statistics.map(stat => <div key={stat.contractId || stat.oracleId}>
```

Comment:
Create two separate components (`OracleStatsView` and `SubscriptionStatsView`) and encapsulate the rendering logic there.